src/host/ncp_host.cpp

```diff
@@ -142,6 +142,7 @@ void NcpHost::Init(void)
     mNcpSpinel.SrpServerSetEnabled(/* aEnabled */ true);
 #endif
 #endif
+    otPlatCryptoRandomInit();
```

Comment:
Why do we need this call? I am not sure we have access to this on the NCP host.

The `otPlat*` APIs are intended for OT stack use. In the NCP architecture, the OT stack would be part of the NCP firmware. On the NCP host (which interacts with the NCP), we should not be able to see or access any of the `otPlat*` APIs.

Reply:
We do, but it is intended for use by the OT stack running on a POSIX platform. It is not intended for use by other components and software modules. We should be careful about creating such dependencies, which do not follow good software design principles and can lead to confusion (some folks may wonder and ask why they would need to implement a subset of `otPlat*` APIs for use on an NCP host software module which is not actually running the OT stack but interacts with an NCP).

If required, the NCP host should define its own set of platform/helper functions it wants to use for things like crypto, random number generation, etc. Reusing the same `otPlat*` APIs is not a good idea.